crates/shared/src/utils.rs

```diff
@@ -323,6 +343,46 @@ mod tests {
             .await
             .expect_err("API is reachable, but is on wrong path");
     }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn event_stream_wrapper_with_idle_timeout() {
+        const TIMEOUT: Duration = Duration::from_secs(3);
+        const IDLE_TIMEOUT: Duration = Duration::from_secs(1);
+
+        let url: Url = format!(
+            "http://localhost:{}",
+            portpicker::pick_unused_port().unwrap()
+        )
+        .parse()
+        .unwrap();
+
+        let app_handle = run_app("hotshot-events", url.clone());
+
+        let mut stream = EventServiceStream::<TestTypes, MockVersion>::connect(url.clone())
+            .await
+            .unwrap();
+
+        // The stream should work when the server is running
+        timeout(TIMEOUT, stream.next())
+            .await
+            .expect("When mock event server is spawned, stream should work")
+            .unwrap();
+
+        // Simulate idle timeout by stopping the server and waiting
+        app_handle.abort();
+        tokio::time::sleep(IDLE_TIMEOUT + Duration::from_millis(500)).await; // Wait longer than idle timeout
```

Comment:
Why can't this just be `IDLE_TIMEOUT`? Or can we increase `IDLE_TIMEOUT` to 2 seconds if we want it to be longer?

Reply:
If these consts are tied (maybe not?) with the setting in non-test code, e.g., the `connect` function, can we move them to the non-test module near `RETRY_PERIOD` and replace hard-coded values with them?

Reply:
500ms is to make sure we wait at least IDLE_TIMEOUT. It's also fine to do a smaller add-on. Updated to `RETRY_PERIOD` in [2a9d3ab](https://github.com/EspressoSystems/marketplace-builder-core/pull/257/commits/2a9d3ab17e82a1bd38b3918a53523b095aee626b).

Reply:
I'm still confused about the purpose of setting the `sleep` period at least `RETRY_PERIOD` if we create a new app immediately afterward.

IIUC when the period exceeds `RETRY_PERIOD`, the `connection.next()` call will enter the `Err(_)` case, whereas before this PR, it would get stuck. However, if we create a `new_app_handle` immediately after this `sleep` period, it looks like we are essentially testing what's already covered in `test_event_stream_wrapper`.

I think we should verify whether `.next()` returns an error as expected after the `sleep`, i.e., whether we do try to reconnect.